src/util/strencodings.cpp

```diff
@@ -526,3 +527,45 @@ std::string HexStr(const Span<const uint8_t> s)
     assert(it == rv.end());
     return rv;
 }
+
+std::optional<uint64_t> ParseByteUnits(const std::string& str, uint64_t default_multiplier)
+{
+    auto multiplier = default_multiplier ? default_multiplier : 1;
```

Comment:
nit, why not `assert(default_multiplier > 0)`?

Reply:
Depends on how strict or how we want to use this function.
For me, a `0` default_multiplier means  "don not apply a multiplier"  if no unit found, parse  `str` as is. This may not be intuitive. 
I'm fine with restricting multiplier > 0.  
@MarcoFalke what's your opinion ? 
either way, I'll update the docs in strencoding.h to explaining what a default_multiplier of zero means.

Reply:
Wouldn't "don not apply a multiplier" mean `1`?

Reply:
ok, i will add an assert